mesonbuild/mdist.py

```diff
@@ -348,6 +355,7 @@ def run(options):
             subprojects[sub] = os.path.join(b.subproject_dir, directory)
         extra_meson_args.append('-Dwrap_mode=nodownload')
 
+    cls: T.Type[Dist]
     if is_git(src_root):
         cls = GitDist
```

Comment:
Not that it matters, but you could have the `Dist` class have a `__new__` method that dispatches to the right implementation class. Something like:
```
class Dist:
    def __new__(...):
        if is_git(src_root):
            return super().__new__(GitDist)
        if is_hg(src_root):
            return super().__new__(HgDist)
        raise ...